auth/src/main/java/com/firebase/ui/auth/provider/IdpProvider.java

```diff
@@ -14,13 +14,15 @@
 
 package com.firebase.ui.auth.provider;
 
+import android.support.annotation.NonNull;
+
 import com.firebase.ui.auth.IdpResponse;
 
 public interface IdpProvider extends Provider {
     interface IdpCallback {
-        void onSuccess(IdpResponse idpResponse);
+        void onSuccess(@NonNull IdpResponse idpResponse);
 
-        void onFailure();
+        void onFailure(@NonNull Exception e);
```

Comment:
Is it possible that anyone was relying on this interface?  If so this is a breaking change .... if this is meant to be internal let's annotate it.

Reply:
I don't think so, or at least, I know we've broken it before without notice. I'll annotate it.